projects/core/src/product/event/product-event.builder.ts

```diff
@@ -49,7 +51,10 @@ export class ProductEventBuilder {
             name: toggled.facetName,
             valueCode: toggled.facetValueCode,
             valueName: toggled.facetValueName,
-            selected: curr.breadcrumbs.length > prev.breadcrumbs.length,
+            selected:
+              curr.breadcrumbs &&
```

Comment:
if `curr.breadcrumbs` is undefined, will `selected` be assigned `false` or `undefined` ?

`selected` is type boolean.

Reply:
`false`. Actually, if `breadcrumbs` is `undefined`. `toggled` is `undefined`. So, the event will not be created.

Reply:
For the kind of code that initialize `selected`, though, there might be something do improve there.  I have a hard time wrapping my head around the result of the expression for some reason, but I tried it in this snipped and it seems that the result of the whole thing can be undefined unless what is in each section is cast into a boolean, see: https://stackblitz.com/edit/typescript-zsm5ev?file=index.ts

Reply:
I am wrong. The value should be `undefined`. The `selected` is `boolean | undefined`. So, it should be fine to assign to `undefined`. That's why it can be compiled.